Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs

```diff
@@ -85,48 +87,75 @@ getAadhaarStatus personId = do
+        return (status, Nothing)
+  return (status, mDriverLicense, message)
 
-getRCAndStatus :: Id SP.Person -> Id DMOC.MerchantOperatingCity -> Int -> Maybe Bool -> Flow (ResponseStatus, Maybe RC.VehicleRegistrationCertificate)
+getRCAndStatus :: Id SP.Person -> Id DMOC.MerchantOperatingCity -> Int -> Maybe Bool -> Flow (ResponseStatus, Maybe RC.VehicleRegistrationCertificate, Maybe Text)
 getRCAndStatus driverId merchantOpCityId onboardingTryLimit multipleRC = do
   associations <- DRAQuery.findAllLinkedByDriverId driverId
   if null associations
     then do
       status <- checkIfInVerification driverId merchantOpCityId onboardingTryLimit Image.VehicleRegistrationCertificate
-      return (status, Nothing)
+      return (status, Nothing, Nothing)
     else do
       mVehicleRCs <- RCQuery.findById `mapM` ((.rcId) <$> associations)
       let vehicleRCs = catMaybes mVehicleRCs
       if isNothing multipleRC -- for backward compatibility
         then do
           let firstRC = listToMaybe vehicleRCs
           case firstRC of
-            Just vehicleRC -> return (mapStatus vehicleRC.verificationStatus, Just vehicleRC)
-            Nothing -> return (NO_DOC_AVAILABLE, Nothing)
+            Just vehicleRC -> do
+              let status = mapStatus vehicleRC.verificationStatus
+              let message = rcVerificationMessage status vehicleRC
+              return (status, Just vehicleRC, message)
+            Nothing -> return (NO_DOC_AVAILABLE, Nothing, Just "No RC document found")
         else do
           let mValidVehicleRC = find (\rc -> rc.verificationStatus == IV.VALID) vehicleRCs
           case mValidVehicleRC of
-            Just validVehicleRC -> return (VALID, Just validVehicleRC)
+            Just validVehicleRC -> return (VALID, Just validVehicleRC, Nothing)
             Nothing -> do
               let mVehicleRC = listToMaybe vehicleRCs
               case mVehicleRC of
-                Just vehicleRC -> return (mapStatus vehicleRC.verificationStatus, Just vehicleRC)
-                Nothing -> return (NO_DOC_AVAILABLE, Nothing)
+                Just vehicleRC -> do
+                  let status = mapStatus vehicleRC.verificationStatus
+                  let message = rcVerificationMessage status vehicleRC
+                  return (status, Just vehicleRC, message)
+                Nothing -> return (NO_DOC_AVAILABLE, Nothing, Just "No RC document found")
 
 mapStatus :: IV.VerificationStatus -> ResponseStatus
 mapStatus = \case
   IV.PENDING -> PENDING
   IV.VALID -> VALID
   IV.INVALID -> INVALID
 
+rcVerificationMessage :: ResponseStatus -> RC.VehicleRegistrationCertificate -> Maybe Text
+rcVerificationMessage status rc = do
+  if status == INVALID || status == FAILED
+    then do
+      let msg = listToMaybe rc.failedRules
+      case msg of
+        Just "id_not_found" -> Just "Wrong RC number entered,Please enter a valid RC number and try again"
+        _ -> Just "Try again after sometime"
+    else Nothing
+
```

Comment:
In `verificationStatus` function, at `Just req` case, add one another case match on `req.status`. If req.status contains `id_not_found` id not found error message and similarly for source_down etc.